packages/terra-form-select/src/shared/_Tag.jsx

```diff
@@ -19,17 +20,58 @@ const propTypes = {
    * The value of the tag.
    */
   value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
+  /**
+   * Specifies whether the tag is disabled.
+   */
+  disabled: PropTypes.bool,
+  /**
+   * @private
+   * The intl object containing translations. This is retrieved from the context automatically by injectIntl.
+   */
+  intl: PropTypes.shape({ formatMessage: PropTypes.func }).isRequired,
 };
 
 /* eslint-disable jsx-a11y/no-static-element-interactions */
-const Tag = ({ children, onDeselect, value }) => {
+const Tag = ({
+  children, onDeselect, value, disabled, intl,
+}) => {
   const theme = React.useContext(ThemeContext);
+  const deselectRef = useRef(null);
+
+  const handleEnterKeyPress = (event) => {
+    if (event.key === 'Enter' && !disabled) {
+      onDeselect(value);
+      const parentUl = deselectRef.current.closest('ul');
+      // Filtering out hidden children
+      const visibleChildrenLis = Array.from(parentUl.children).filter(child => child.tagName === 'LI' && !/visually-hidden-component/.test(child.className));
+      if (visibleChildrenLis.length >= 2) {
+        const secondLastLi = visibleChildrenLis[visibleChildrenLis.length - 2];
+        if (secondLastLi) {
+          const spanChildren = secondLastLi.children;
+          if (spanChildren[1]) {
+            spanChildren[1].focus();
+          }
+        }
+      } else {
+        parentUl.click();
+      }
+    }
+  };
   return (
     <li className={cx('tag', theme.className)}>
       <span className={cx('display')}>
         {children}
       </span>
-      <span className={cx('deselect')} onClick={() => { onDeselect(value); }} role="presentation">
+      <span
+        ref={deselectRef}
+        onKeyDown={handleEnterKeyPress}
+        className={cx('deselect')}
+        onClick={() => { if (!disabled) onDeselect(value); }}
+        // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
+        tabIndex={!disabled ? 0 : null}
+        role={!disabled ? 'button' : null}
+        aria-label={!disabled ? `${intl.formatMessage({ id: 'Terra.form.select.deselect' })} ${children}` : null}
```

Comment:
```suggestion
        aria-label={!disabled ? `intl.formatMessage({ id: 'Terra.form.select.deselect' }, { text: stringValue });` : null}
```
Any js value should be passed in the above manner to formatMessage and you should not be passing a node(children) to it. Pass a string value